### PR TITLE
search: update error message for structural query alert

### PR DESCRIPTION
**Pull request by leonore**
found as part of [this slack thread](https://sourcegraph.slack.com/archives/C03CSAER9LK/p1653666884809919) on #feedback-dogfood that the alert for a structural search does not make much sense in an insights context as we don't have a UI toggle for patterntype:structural.

this updates the error message so that it might be a bit more useful in those cases

## Test plan
also pushed to a backend-integration branch.

validated the new error in an insight locally: 
Screenshot 2022-05-30 at 17 52 10

using the query: `<Tooltip...content...>...</Tooltip>`

### CODE REVIEW

File: dev/gqltest/search_test.go [hunk 1]
```diff
@@ -717,7 +717,7 @@ func testSearchClient(t *testing.T, client searchClient) {
 				zeroResult: true,
 				wantAlert: &gqltestutil.SearchAlert{
 					Title:       "No results",
-					Description: "It looks like you may have meant to run a structural search, but it is not toggled.",
+					Description: "It looks like you may have meant to run a structural search, but the patterntype is not specified or it is not toggled.",
```
Comment (lrhacker) [inline dev/gqltest/search_test.go:720]:
```suggestion
					Description: "It looks like you're trying to run a structural search, but it was not enabled using the patterntype keyword or UI toggle.",
```

Being clear about a search keyword vs a UI element might make it easier and/or quicker for folks to figure out how to resolve the issue. Without saying what the patterntype value is, a user would still need to look up docs, but being able to look for "keyword patterntype" might more quickly find the solution.

Are there any copy guidelines for whether to communicate the exact resolution to a user? For example, adding on: "Please enable structural search and try again." I know I've seen general copy guidelines, but don't recall seeing anything around error messages specifically (but I'm also still learning and could have missed something!). Not sure if design would have input here. :)

Comment (leonore) [reply dev/gqltest/search_test.go]: Thanks for the better words, they're not my strong suit. I haven't seen any guidelines before, but I can ask tomorrow

File: dev/gqltest/search_test.go [hunk 2]
```diff
@@ -732,7 +732,7 @@ func testSearchClient(t *testing.T, client searchClient) {
...
```
Comment (lrhacker) [inline dev/gqltest/search_test.go:735]:
```suggestion
					Description: "It looks like you're trying to run a structural search, but it is not enabled using the patterntype keyword or UI toggle.",
```

Copied the same suggestion to the other lines 😄

File: internal/search/alert.go [hunk 3]
```diff
@@ -140,7 +140,7 @@ func AlertForStructuralSearchNotSet(queryString string) *Alert {
 	return &Alert{
 		PrometheusType: "structural_search_not_set",
 		Title:          "No results",
-		Description:    "It looks like you may have meant to run a structural search, but it is not toggled.",
+		Description:    "It looks like you may have meant to run a structural search, but the patterntype is not specified or it is not toggled.",
```
Comment (lrhacker) [inline internal/search/alert.go:143]:
```suggestion
		Description:    "It looks like you're trying to run a structural search, but it is not enabled using the patterntype keyword or UI toggle.",
```

Copied the same suggestion to the other lines 😄

File: dev/gqltest/search_test.go (hunk 1 above)
Comment (lrhacker) [reply dev/gqltest/search_test.go]: I usually default to asking design to proof read user-facing copy to make sure it has the seal of approval to fit in with the rest of the copy in the app they've put together. It's really hard to pick words sometimes 😅

File: internal/search/alert.go (hunk 3 above)
Comment (leonore) [reply internal/search/alert.go]: thanks for catching that, did this too late in my day

**Review by camdencheek — APPROVED** [review]
Suggestion wording looks good to me!

**Pull request merged**